js/base/Exchange.js

```diff
@@ -819,6 +819,108 @@ module.exports = class Exchange {
         throw new NotSupported (this.id + ' fetchPermissions() not supported yet')
     }
 
+    getInitialPermissions () {
+        const broadPropertyNames = {
+            'publicAPI': 0,
+            'privateAPI': 0,
+            'CORS': 0,
+            'spot': 0,
+            'margin': 0,
+            'swap': 0,
+            'future': 0,
+            'option': 0,
+        };
+        // public endpoints, which are considered to work without need of api-key (though, in some exchanges some 'public' endpoints will be manually overriden as 'non-public')
+        const publicPropertyNames = {
+            'fetchBidsAsks': 0,
+            'fetchCurrencies': 0,
+            'fetchL2OrderBook': 0,
+            'fetchLeverageTiers': 0,
+            'fetchMarketLeverageTiers': 0,
+            'fetchMarkets': 0,
+            'fetchMarkOHLCV': 0,
+            'fetchOHLCV': 0,
+            'fetchOrderBook': 0,
+            'fetchOrderBooks': 0,
+            'fetchPremiumIndexOHLCV': 0,
+            'fetchStatus': 0,
+            'fetchTicker': 0,
+            'fetchTickers': 0,
+            'fetchTime': 0,
+            'fetchTrades': 0,
+            'fetchTradingFee': 0,
+            'fetchTradingFees': 0,
+            'fetchTradingLimits': 0,
+            'loadMarkets': 0,
+            'fetchBorrowInterest': 0,
+            'fetchBorrowRate': 0,
+            'fetchBorrowRatesPerSymbol': 0,
+            'fetchBorrowRates': 0,
+            'fetchFundingFee': 0,
+            'fetchFundingFees': 0,
+            'fetchFundingRate': 0,
+        };
+        const publiclyAvailables = Object.assign (broadPropertyNames, publicPropertyNames);
+        const initialPermissions = {};
+        const allKeys = Object.keys (this.has);
+        for (let i = 0; i < allKeys.length; i++) {
+            const methodName = allKeys[i];
+            if (methodName in publiclyAvailables) {
+                // if method is non-private endpoint, set it to same value as in `.has`
+                initialPermissions[methodName] = this.has[methodName];
+            } else {
+                // if method is private endpoint, then set it to undefined, to be overriden by implementation
+                initialPermissions[methodName] = undefined;
+            }
+        }
+        // if user didn't set api-keys, then we can't further check private endpoints, thus return 
+        const apikeySet = this.checkRequiredCredentials (false) === true;
+        this.initialPermissions = initialPermissions;
+        return [apikeySet, initialPermissions];
+    }
+
+    safePermissions (permissionsMap = {}) {
```

Comment:
I believe, the purpose of `safePermissions` must be to take an incomplete `has`-like dictionary in its argument and return a complete `has`-like dictionary.

```JavaScript
safePermissions ({ 'createOrder': true }) // no more fields here
// would return
{
    'CORS': exchange.has['CORS'],
    'spot': exchange.has['spot'],  
    'swap': exchange.has['swap'],
    // all other fields here
}
```

Reply:
> but real-world examples would not be such

I would suggest to use the real-world examples as our base ground for the design of this aspect. It would be a bit easier to discuss if you paste a few such examples here. In general at this time I still don't think we should introduce things like `trade`, `funds` or `transfer` in this PR.

Reply:
I will post an example here as soon as I return to `fetchPersmissions` task. I think there was probably some miscomunication from my side, as we will not be introducing anything (`trade`, `funds`, `transfer` or whtever), they are just exchange-specific keys. 
I will try to provide a good examples and you will be convinced too that the way I suggest, will not be bad way, and instead possibly best way for given task (please allow me a while before I reach this task back again, before we reject the suggested approach).